tests/kafkatest/utils/security_config.py

```diff
@@ -0,0 +1,132 @@
+
+        return {
+            'ssl.keystore.location' : ksPath,
+            'ssl.keystore.password' : ksPassword,
+            'ssl.key.password' : keyPassword,
+            'ssl.truststore.location' : tsPath,
+            'ssl.truststore.password' : tsPassword
+        }
+
+    @staticmethod
+    def runcmd(cmd):
+        proc = subprocess.Popen(cmd, shell=True, stdout=subprocess.PIPE, stderr=subprocess.STDOUT)
+        proc.communicate()
+        if proc.returncode != 0:
+            raise subprocess.CalledProcessError(proc.returncode, cmd)
+
+
+class SecurityConfig:
+
+    PLAINTEXT = 'PLAINTEXT'
+    SSL = 'SSL'
+    SSL_DIR = "/mnt/ssl"
+    KEYSTORE_PATH = "/mnt/ssl/test.keystore.jks"
+    TRUSTSTORE_PATH = "/mnt/ssl/test.truststore.jks"
+
+    ssl_stores = Keytool.generate_keystore_truststore('.')
+
+    def __init__(self, node_account, security_protocol, template_props=""):
+        """
+        Initialize the security properties for the node and copy
+        keystore and truststore to the remote node if the transport protocol 
+        is SSL. If security_protocol is None, the protocol specified in the
+        template properties file is used. If no protocol is specified in the
+        template properties either, PLAINTEXT is used as default.
+        """
+
+        self.node_account = node_account
+        if security_protocol is None:
+            for line in template_props.split("\n"):
+                items = line.split("=")
+                if len(items) == 2 and items[0].strip() == 'security.protocol':
+                    security_protocol = str(items[1].strip())
+                    if security_protocol not in [SecurityConfig.PLAINTEXT, SecurityConfig.SSL]:
+                        raise Exception("Invalid security.protocol in template properties: " + security_protocol)
+        if security_protocol is None:
+            security_protocol = SecurityConfig.PLAINTEXT
+        if security_protocol == SecurityConfig.SSL:
+            node_account.ssh("mkdir -p %s" % SecurityConfig.SSL_DIR, allow_fail=False)
+            node_account.scp_to(SecurityConfig.ssl_stores['ssl.keystore.location'], SecurityConfig.KEYSTORE_PATH)
+            node_account.scp_to(SecurityConfig.ssl_stores['ssl.truststore.location'], SecurityConfig.TRUSTSTORE_PATH)
+
+        self.properties = {
+            'security.protocol' : security_protocol,
+            'ssl.keystore.location' : SecurityConfig.KEYSTORE_PATH,
+            'ssl.keystore.password' : SecurityConfig.ssl_stores['ssl.keystore.password'],
+            'ssl.key.password' : SecurityConfig.ssl_stores['ssl.key.password'],
+            'ssl.truststore.location' : SecurityConfig.TRUSTSTORE_PATH,
+            'ssl.truststore.password' : SecurityConfig.ssl_stores['ssl.truststore.password']
+        }
+    
+    def write_to_file(self, prop_file='/mnt/ssl/security.properties'):
+        """Write security properties to a remote file"""
+        tmpfile =  tempfile.NamedTemporaryFile(delete=False)
```

Comment:
There's a utility in ducktape on `RemoteAccount` for doing this already called `create_file`. Given a string with the contents, it'll generate the tempfile locally, scp it to the destination, and then clean up the tempfile.